airflow/providers/amazon/aws/hooks/emr.py

```diff
@@ -126,14 +126,21 @@ def create_job_flow(self, job_flow_overrides: dict[str, Any]) -> dict[str, Any]:
         return response
 
     def add_job_flow_steps(
-        self, job_flow_id: str, steps: list[dict] | str | None = None, wait_for_completion: bool = False
+        self,
+        job_flow_id: str,
+        steps: list[dict] | str | None = None,
+        wait_for_completion: bool = False,
+        waiter_delay: int = 5,
+        waiter_max_attempts: int = 100,
```

Comment:
Hmm.. what do you think if by default `waiter_delay` and `waiter_max_attempts` set to `None`  and use in waiter config only if this values not None?

In this case default values from `botocore` waiters definition file will use: delay=30, max_attempts=60.

Something like that

```python
waiter.wait(
    ClusterId=job_flow_id,
    StepId=step_id,
    WaiterConfig=trim_none_values(
        {
            "Delay": waiter_delay,
            "MaxAttempts": waiter_max_attempts,
        }
    ),
)
```

Reply:
yes, I considered options, but chose to keep it as closer to author's decision.
Maybe even better option is to pass config as a parameter
```
def add_job_flow_steps(
        self, job_flow_id: str, steps: list[dict] | str | None = None, wait_for_completion: bool = False
        self,
        job_flow_id: str,
        steps: list[dict] | str | None = None,
        wait_for_completion: bool = False,
        waiter_config: dict | None = None,
```

Reply:
Yep, in this case we reduce number of parameters.

But we need to be sure that we send as a dictionary (otherwise it raise an error) and only `Delay` and `MaxAttempts` are acceptable parameters.

Unfortunetly waiters not well documented or better say `botocore` sources are only the documentation of waiters internals:

https://github.com/boto/botocore/blob/6af5c29d9c71deb325eca99f24698303fc9e3dea/botocore/waiter.py#L336-L338